File: epi_judge_java/epi/MaxTrappedWater.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.List;

public class MaxTrappedWater {

    @EpiTest(testDataFile = "max_trapped_water.tsv")
    public static int getMaxTrappedWater(List<Integer> heights) {
        int left = 0;
        int right = heights.size() - 1;

        int result = 0;

        while (left < right) {
            result = Math.max(result, Math.min(heights.get(left), heights.get(right)) * (right - left));
            if (heights.get(left) < heights.get(right)) {
                left++;
            } else {
                right--;
            }
        }

        return result;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "MaxTrappedWater.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/max_trapped_water.cc
#include <vector>

#include "test_framework/generic_test.h"
using namespace std;

int GetMaxTrappedWater(const vector<int> &heights) {
    int left = 0;
    int right = heights.size() - 1;
    int ans = 0;
    while (left < right) {
        ans = max(ans, (right - left) * min(heights[left], heights[right]));
        if (heights[left] < heights[right]) left++;
        else right--;
    }
    return ans;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"heights"};
    return GenericTestMain(args, "max_trapped_water.cc", "max_trapped_water.tsv",
                           &GetMaxTrappedWater, DefaultComparator{}, param_names);
}
